# NDS Data Quality Check

## Row counts comparison between NDS and Stage

In [1]:
-- Conbine row count from tables of Stage and NDS
SELECT table_name,[NDS],[Stage], 
	   COALESCE([NDS],0) - COALESCE([Stage],0) AS nds_Stage_diff
FROM (
	SELECT 
		  'Stage' AS db,t.name AS table_name,
		   SUM(p.row_count) AS row_count
		FROM 
	  Stage.sys.tables AS t
	  INNER JOIN 
		(SELECT ss.object_id, SUM(ss.row_count) AS row_count 
		 FROM Stage.sys.dm_db_partition_stats ss
		 WHERE ss.index_id < 2
		 GROUP BY ss.object_id) AS p 
	  ON t.object_id = p.object_id
	GROUP BY t.name
	UNION
	SELECT 
	  'NDS' AS db,t.name AS table_name,
	  SUM(p.row_count) AS row_count
	FROM 
	  NDS.sys.tables AS t
	  INNER JOIN 
		(SELECT ss.object_id, SUM(ss.row_count) AS row_count 
		 FROM NDS.sys.dm_db_partition_stats ss
		 WHERE ss.index_id < 2
		 GROUP BY ss.object_id) AS p 
	  ON t.object_id = p.object_id
	GROUP BY t.name
	) as src
	PIVOT(SUM(row_count)
	  FOR db IN (NDS,Stage)
	  ) AS piv


(59 rows affected)

Total execution time: 00:00:01.367

table_name NDS Stage nds_Stage_diff address 108412 NULL 108412 address_junction 49592 NULL 49592 address_type 2 2 0 artist 597 576 21 business_unit 0 NULL 0 campaign 0 NULL 0 campaign_result 0 NULL 0 channel 6 NULL 6 channel_junction 108316 NULL 108316 city 341 NULL 341 communication 0 NULL 0 communication_category 0 NULL 0 communication_subscription 0 NULL 0 contact 0 NULL 0 country 243 243 0 currency 175 175 0 currency_rate 121 121 0 customer 54158 54158 0 customer_status 4 4 0 customer_type 3 3 0 delivery_status 0 NULL 0 delivery_status_category 0 NULL 0 division 3 3 0 email_address 54155 NULL 54155 email_address_junction 54158 NULL 54158 email_address_type 2 2 0 format 28 27 1 household_income 11 11 0 interest 38 38 0 interest_junction 160351 NULL 160351 language 0 505 -505 lead 0 NULL 0 media 6 NULL 6 occupation 101 101 0 order_detail NULL 63778 -63778 order_header 44425 44425 0 order_item 63770 NULL 63770 package 0 24 -24 package_type 0 6 -6 permission 2 2 0 phone_number 52822 NULL 52822 phone_number_junction 54158 NULL 54158 phone_number_type 2 2 0 product 2459 2458 1 product_category 4 3 1 product_status 6 4 2 product_type 66 65 1 region 11 11 0 state 50 50 0 store 102 102 0 store_type 4 NULL 4 subscriber_band 4 NULL 4 subscriber_class 4 NULL 4 subscription_sales 0 NULL 0 subscription_status 0 NULL 0 supplier 0 NULL 0 supplier_performance 0 NULL 0 supplier_status 0 NULL 0 sysdiagrams 5 NULL 5

## Test 2

|table_name	|NDS	|Stage	|nds_Stage_diff|Note|
|---|---|---|---|---|
|address	|108412	|NULL	|108412|
|address_junction	|49592	|NULL	|49592|
|address_type	|2	|2	|0|
|artist	|597	|576	|21| From Jade Artist2|
|business_unit	|0	|NULL	|0|
|campaign	|0	|NULL	|0|
|campaign_result	|0	|NULL	|0|
|channel	6	|NULL	|6|
|channel_junction	|108316	|NULL	|108316|
|city	|341	|NULL	|341|
|communication	0	NULL	0
|communication_category	0	NULL	0
communication_subscription	0	NULL	0
contact	0	NULL	0
country	243	243	0
currency	175	175	0
currency_rate	121	121	0
customer	54158	54158	0
customer_status	4	4	0
customer_type	3	3	0
delivery_status	0	NULL	0
delivery_status_category	0	NULL	0
division	3	3	0
email_address	54155	NULL	54155
email_address_junction	54158	NULL	54158
email_address_type	2	2	0
format	28	27	1
household_income	11	11	0
interest	38	38	0
interest_junction	160351	NULL	160351
language	0	505	-505
lead	0	NULL	0
media	6	NULL	6
occupation	101	101	0
order_detail	NULL	63778	-63778
order_header	44425	44425	0
order_item	63770	NULL	63770
package	0	24	-24
package_type	0	6	-6
permission	2	2	0
phone_number	52822	NULL	52822
phone_number_junction	54158	NULL	54158
phone_number_type	2	2	0
product	2459	2458	1
product_category	4	3	1
product_status	6	4	2
product_type	66	65	1
region	11	11	0
state	50	50	0
store	102	102	0
store_type	4	NULL	4
subscriber_band	4	NULL	4
subscriber_class	4	NULL	4
subscription_sales	0	NULL	0
subscription_status	0	NULL	0
supplier	0	NULL	0
supplier_performance	0	NULL	0
supplier_status	0	NULL	0
sysdiagrams	5	NULL	5